src/compiler/checker.ts

```diff
@@ -12616,6 +12616,13 @@ namespace ts {
                         }
                     }
                     checkKindsOfPropertyMemberOverrides(type, baseType);
+                        
+                    // Check if base type declaration appears before heritage clause to avoid false errors for 
+                    // base type declarations in the extend clause itself
+                    let heritageClassElement = getClassExtendsHeritageClauseElement(node);

```

Comment:
Fixed this part. Still need to fix the part where I will report error on alias declaration:
```typescript
import a = M.B; // error
import b = M.I; // no error
class C extends a { // no error 
}
class D extends M.B { // error
}
module M {
    export class B {
    }
    export interface I {
    }
}
```